cms/static/js/utils/date_utils.js

```diff
@@ -1,15 +1,80 @@
+function($, date, TriggerChangeEventOnEnter) {
+    'use strict';
+    var setupDatePicker = function (fieldName, view, index) {
+        var cacheModel;
+        var div;
+        if (typeof index !== "undefined" && view.hasOwnProperty("collection")) {
+            cacheModel = view.collection.models[index];
+            div = view.$el.find('#' + view.collectionSelector(cacheModel.cid));
+        } else {
+            cacheModel = view.model;
+            div = view.$el.find('#' + view.fieldToSelectorMap[fieldName]);
+        }
+        var datefield = $(div).find("input.date");
+        var timefield = $(div).find("input.time");
+        var cacheview = view;
+        var setfield = function () {
+            var newVal = getDate(datefield, timefield),
+                oldTime = new Date(cacheModel.get(fieldName)).getTime();
+            if (newVal) {
+                if (!cacheModel.has(fieldName) || oldTime !== newVal.getTime()) {
+                    cacheview.clearValidationErrors();
+                    cacheview.setAndValidate(fieldName, newVal);
+                }
+            }
+            else {
+                // Clear date (note that this clears the time as well, as date and time are linked).
+                // Note also that the validation logic prevents us from clearing the start date
+                // (start date is required by the back end).
+                cacheview.clearValidationErrors();
+                cacheview.setAndValidate(fieldName, null);
+            }
+        };
+
+        // instrument as date and time pickers
+        timefield.timepicker({'timeFormat' : 'H:i'});
+        datefield.datepicker();
+
+        // Using the change event causes setfield to be triggered twice, but it is necessary
+        // to pick up when the date is typed directly in the field.
+        datefield.change(setfield).keyup(TriggerChangeEventOnEnter);
+        timefield.on('changeTime', setfield);
+        timefield.on('input', setfield);
+
+        var current_date = null;
+        if (cacheModel) {
+            current_date = cacheModel.get(fieldName);
+        }
+        // timepicker doesn't let us set null, so check that we have a time
+        if (current_date) {
+            setDate(datefield, timefield, current_date);
+        } // but reset fields either way
+        else {
+            timefield.val('');
+            datefield.val('');
+        }
+    };
+
     var getDate = function (datepickerInput, timepickerInput) {
         // given a pair of inputs (datepicker and timepicker), return a JS Date
         // object that corresponds to the datetime.js that they represent. Assume
         // UTC timezone, NOT the timezone of the user's browser.
-        var date = $(datepickerInput).datepicker("getDate");
-        var time = $(timepickerInput).timepicker("getTime");
+        var date = null, time = null;
+        if (datepickerInput) {
```

Comment:
When is datepickerInput ever falsey? If it is legitimate for it to not be present, it seems that you would need to do the same length check that you do for timepickerInput. But if it really never happens, let's just remove the check.